import{BugIndicatingError as r,onUnexpectedError as t}from"./errors.js";function l(e,n){if(!e)throw new Error(n?`Assertion failed (${n})`:"Assertion Failed")}function f(e,n="Unreachable"){throw new Error(n)}function c(e,n="unexpected state"){if(!e)throw new r(`Assertion Failed: ${n}`)}function u(e){e||t(new r("Soft Assertion Failed"))}function d(e){if(!e()){debugger;e(),t(new r("Assertion Failed"))}}function w(e,n){let o=0;for(;o<e.length-1;){const i=e[o],s=e[o+1];if(!n(i,s))return!1;o++}return!0}export{c as assert,d as assertFn,f as assertNever,w as checkAdjacentItems,l as ok,u as softAssert};
